### DISCUSSION TITLE
RAM Disk Speed SLOWER than NVMe 4.0? PLZ HELP!!!

### DISCUSSION THREAD

**Discussion by RotiCanaiX** [madMAx43v3r/chia-plotter#817] [Q&A]
Hi all, hope some gurus can enlighten me.

I have the following system:
Intel i7-11700
Asrock B560 Steel Legend
128GB @ 32GBx4 DDR4 3200 RAM @ 16-20-20-38 1T (previously 64GB @ 16GBx4 b4 upgrade)
Corsair MP600 1TB x 3 units m.2 slots (1 for OS, 1 running PCIe 4.0, 1 running PCIe 3.0)
Samsung PM981 2TB x2 m.2 (connected using PCIe to M.2 adapter) running as final drive b4 robocopy to farmer drive
Windows 10 Pro/ Windows 11 Pro (dev build) - updated windows updates always, latest available drivers

Drive speed (read/write) Sequential via Crystal Disk Mark
Corsair MP600 PCIe 4.0 4.9GBps/4.2GBps
Corsair MP600 PCIe 3.0 3.3GBps/3.2GBps
Samsung PM981 3.3GBps/ 2.4GBps
RAM Drive via OSF Disk 9.9GBps/ 9.8GBps

**### The question:**

I get the exact similar timing if I use RAM Disk as Tmp2 or use PCIe 4.0 as Tmp2 which is about 32-36mins. 
Can someone enlighten me what is the reason?
I know jumping to Linux could be faster but I am not technical and have been Windows my life.
My test results clearly shows the RAM drive being faster but actual plotting shows the same.
I have tried all the stotiks plotter v.004, 00.5, 00.6, 0.07 and latest 0.1.2a with all similar results.

### **_my plotting with NVMe ONLY (tmp1 and tmp2)_**

Final Directory: Samsung PM981
Number of Plots: 1
Crafting plot 1 out of 1
Process ID: 15228
Number of Threads: 32
Number of Buckets P1:    2^8 (256)
Number of Buckets P3+P4: 2^8 (256)
Pool Public Key:   XXX
Farmer Public Key: XXX
Working Directory:   PCIe 3.0 NVMe
Working Directory 2: PCIe 4.0 NVMe
Plot Name: plot-k32-2021-06-23-18-04-5f7cb33da5213c506645f994baa6f1b683e2c2e2043bf1f443637b249d39fff1
[P1] Table 1 took 16.7495 sec
[P1] Table 2 took 122.552 sec, found 4294931324 matches
[P1] Table 3 took 152.657 sec, found 4294851622 matches
[P1] Table 4 took 193.843 sec, found 4294646882 matches
[P1] Table 5 took 187.955 sec, found 4294230081 matches
[P1] Table 6 took 177.566 sec, found 4293440844 matches
[P1] Table 7 took 127.336 sec, found 4291882422 matches
Phase 1 took 978.865 sec
[P2] max_table_size = 4294967296
[P2] Table 7 scan took 7.43274 sec
[P2] Table 7 rewrite took 21.6428 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 31.9004 sec
[P2] Table 6 rewrite took 50.7728 sec, dropped 581464806 entries (13.5431 %)
[P2] Table 5 scan took 30.79 sec
[P2] Table 5 rewrite took 48.0217 sec, dropped 762203020 entries (17.7495 %)
[P2] Table 4 scan took 30.7505 sec
[P2] Table 4 rewrite took 47.3215 sec, dropped 829034744 entries (19.3039 %)
[P2] Table 3 scan took 30.5685 sec
[P2] Table 3 rewrite took 47.1506 sec, dropped 855199547 entries (19.9122 %)
[P2] Table 2 scan took 30.1458 sec
[P2] Table 2 rewrite took 46.7796 sec, dropped 865637346 entries (20.1549 %)
Phase 2 took 435.096 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 37.5305 sec, wrote 3429293978 right entries
[P3-2] Table 2 took 30.6918 sec, wrote 3429293978 left entries, 3429293978 final
[P3-1] Table 3 took 52.1997 sec, wrote 3439652075 right entries
[P3-2] Table 3 took 32.2197 sec, wrote 3439652075 left entries, 3439652075 final
[P3-1] Table 4 took 52.5458 sec, wrote 3465612138 right entries
[P3-2] Table 4 took 31.5473 sec, wrote 3465612138 left entries, 3465612138 final
[P3-1] Table 5 took 53.5826 sec, wrote 3532027061 right entries
[P3-2] Table 5 took 32.2245 sec, wrote 3532027061 left entries, 3532027061 final
[P3-1] Table 6 took 56.4041 sec, wrote 3711976038 right entries
[P3-2] Table 6 took 33.7977 sec, wrote 3711976038 left entries, 3711976038 final
[P3-1] Table 7 took 44.1988 sec, wrote 4291882422 right entries
[P3-2] Table 7 took 38.9998 sec, wrote 4291882422 left entries, 4291882422 final
Phase 3 took 499.416 sec, wrote 21870443712 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 64.2126 sec, final plot size is 108794578882 bytes
Total plot creation time was 1977.67 sec (32.9612 min)

### **_my plotting with PCIe 4.0 NVMe tmp1 and RAM Disk via OSF Disk tmp2_**

Final Directory: Samsung PM981
Number of Plots: infinite
Crafting plot 1 out of -1
Process ID: 9440
Number of Threads: 32
Number of Buckets P1:    2^8 (256)
Number of Buckets P3+P4: 2^8 (256)
Pool Public Key: XXX  
Farmer Public Key: XXX
Working Directory:   PCIe 4.0 NVMe
Working Directory 2: RAM Disk via OSF Disk 110GB
Plot Name: plot-k32-2021-07-13-22-09-13daa1aa9b01a9410cbfb58c3aedbbab5ac7d0ab06c108c30c08a9374c90d240
[P1] Table 1 took 18.6852 sec
[P1] Table 2 took 130.747 sec, found 4295043599 matches
[P1] Table 3 took 168.128 sec, found 4295016742 matches
[P1] Table 4 took 214.456 sec, found 4295086393 matches
[P1] Table 5 took 209.425 sec, found 4295107835 matches
[P1] Table 6 took 196.845 sec, found 4294988650 matches
[P1] Table 7 took 142.553 sec, found 4294954688 matches
Phase 1 took 1081.06 sec
[P2] max_table_size = 4295107835
[P2] Table 7 scan took 10.1059 sec
[P2] Table 7 rewrite took 25.9555 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 35.8062 sec
[P2] Table 6 rewrite took 59.0091 sec, dropped 581279129 entries (13.5339 %)
[P2] Table 5 scan took 32.5453 sec
[P2] Table 5 rewrite took 52.9807 sec, dropped 762094679 entries (17.7433 %)
[P2] Table 4 scan took 33.972 sec
[P2] Table 4 rewrite took 55.0679 sec, dropped 828943441 entries (19.2998 %)
[P2] Table 3 scan took 31.8503 sec
[P2] Table 3 rewrite took 51.4958 sec, dropped 855082187 entries (19.9087 %)
[P2] Table 2 scan took 34.3904 sec
[P2] Table 2 rewrite took 56.7888 sec, dropped 865590473 entries (20.1532 %)
Phase 2 took 482.752 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 44.8304 sec, wrote 3429453126 right entries
[P3-2] Table 2 took 37.124 sec, wrote 3429453126 left entries, 3429453126 final
[P3-1] Table 3 took 58.3739 sec, wrote 3439934555 right entries
[P3-2] Table 3 took 38.1144 sec, wrote 3439934555 left entries, 3439934555 final
[P3-1] Table 4 took 59.0791 sec, wrote 3466142952 right entries
[P3-2] Table 4 took 38.5468 sec, wrote 3466142952 left entries, 3466142952 final
[P3-1] Table 5 took 60.0628 sec, wrote 3533013156 right entries
[P3-2] Table 5 took 38.8195 sec, wrote 3533013156 left entries, 3533013156 final
[P3-1] Table 6 took 62.6846 sec, wrote 3713709521 right entries
[P3-2] Table 6 took 41.8903 sec, wrote 3713709521 left entries, 3713709521 final
[P3-1] Table 7 took 53.1172 sec, wrote 4294954688 right entries
[P3-2] Table 7 took 47.4162 sec, wrote 4294954688 left entries, 4294954688 final
Phase 3 took 581.146 sec, wrote 21877207998 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 57.8331 sec, final plot size is 108835712471 bytes
Total plot creation time was 2202.85 sec (36.7142 min)

**Comment by ataa** [comment]
![Screenshot_2](https://user-images.githubusercontent.com/908955/125485573-c735cafa-db3c-4f91-9aff-441f2e378896.jpg)

**Reply by RotiCanaiX** [reply]
hi, are u suggesting that OSF is slower?
can u send me the link for the above image?
thanks for replying.

**Reply by RotiCanaiX** [reply]
I just installed StarWind and is trying out now. The Initial [P1] Table 1 is 19 seconds, which I believe means it still slower. Anyway, let me run a a few passes and update. Thanks

**Comment by ataa** [comment]
I think so, they did the test on  Intel ® Core i5-4460 3,2GHz with 8 GB DDR3 1600MHz using an older version of OSFMount, but I think it may be the issue here. Also It could be Windows Virtual Memory.
https://www.starwindsoftware.com/blog/ram-disk-technology-performance-comparison

**Reply by RotiCanaiX** [reply]
Windows Virtual Memory as in disable Virtual Memory caching into the disk? The 128GB is still fine I believe as there are still 16GB left for everything else. Anyway, will try disable disk caching. Hope it doesnt crash windows.

**Comment by RotiCanaiX** [comment]
@madMAx43v3r can u please help enlighten? I have tried many RAM Disk software yesterday but to no avail. issues still persist. Or is there a speed cap for the plotter? meaning anything above something like 4GBps doesnt help time? Thanks

**Reply by gryan315** [reply]
Don't use windows, use any linux distro with a mainline kernal and create ramdisk with tmpfs.

**Reply by RotiCanaiX** [reply]
does it mean ramdisk in Linux is for sure faster? If yes, i will study up some Linux and how to do Chia.

**Reply by gryan315** [reply]
Yes Linux is faster. If you're going to use Ubuntu, install 21.04 to get a more up to date kernel, and make sure the Linux CPU governor is set to "performance".
